### PR TITLE
use pmcd pointer for immutable graphs

### PR DESCRIPTION
**Pull request by MauricioUyaguari**
## Summary
- [x] Rename `V1_AlloySDLC` -> `V1_LegendSDLC`
- [x] Add sdlc pointer on `PureModel`
- [x] Leverage https://github.com/finos/legend-depot for `execute` calls on an immutable graph (legend query). This means we will send a pmcd pointer for execute calls instead of the full pmcd. This helps us as we will leverage `engine`'s already implemented caching of pmcd when calling metadata.

This PR main concern is in helping address https://github.com/finos/legend-studio/issues/1903 and https://github.com/finos/legend-studio/issues/1904 by leveraging pmcd pointer. We have migrated the execute api to take in both pmcd and pmcd pointer (changing V1_PureModelContextData to V1_PuremodelContext). We will migrate the rest of the apis following testing of this as well as confirmation engine uses the correct load pure model function on those apis to correctly accept pmcd and pmcd pointer. 

## How did you test this change?

- [ ] Test(s) added
- [ ] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

File: packages/legend-graph/src/graph/BasicModel.ts [hunk 1]
```diff
@@ -99,6 +100,9 @@ const FORBIDDEN_EXTENSION_ELEMENT_CLASS = new Set([
  */
 export abstract class BasicModel {
   root: Package;
+
+  private _sdlcPointer: LegendSDLC | undefined;
```
Comment (akphi) [inline packages/legend-graph/src/graph/BasicModel.ts:104]: `private readonly`

File: packages/legend-application-studio/src/stores/project-viewer/ProjectViewerStore.ts [hunk 2]
```diff
@@ -387,12 +388,22 @@ export class ProjectViewerStore {
...
+          )
+        : undefined;
       // build graph
       const graph_buildReport =
         (yield this.editorStore.graphManagerState.graphManager.buildGraph(
           this.editorStore.graphManagerState.graph,
           entities,
           this.editorStore.graphManagerState.graphBuildState,
+          {
+            sdlc,
```
Comment (akphi) [inline packages/legend-application-studio/src/stores/project-viewer/ProjectViewerStore.ts:405]: just inline is fine

File: packages/legend-storage/src/SDLC.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ProjectGAVCoordinates } from './DependencyUtils.js';
+
+export class LegendSDLC implements ProjectGAVCoordinates {
```
Comment (akphi) [inline packages/legend-storage/src/SDLC.ts:19]: this file being here looks wrong to me, perhaps let's have an abstract file called `GraphDataOrigin` and make `LegendSDLC extends` that, we don't even need to say `implements ProjectGAVCoordinates` doesn't buy us any good

File: packages/legend-graph/src/graph/BasicModel.ts [hunk 4]
```diff
@@ -208,6 +212,20 @@ export abstract class BasicModel {
     return [...this.ownServices, ...this.ownMappings];
   }
 
+  get sdlc(): LegendSDLC | undefined {
```
Comment (akphi) [inline packages/legend-graph/src/graph/BasicModel.ts:215]: --> `origin`

File: packages/legend-graph/src/graph/BasicModel.ts [hunk 5]
```diff
@@ -208,6 +212,20 @@ export abstract class BasicModel {
...
+    return this._sdlcPointer;
+  }
+
+  setSDLC(val: LegendSDLC): void {
```
Comment (akphi) [inline packages/legend-graph/src/graph/BasicModel.ts:219]: `setOrigin`

**Review by akphi — CHANGES_REQUESTED** [review]
left some comments for cleanup in future PR ;)

**Review by akphi — APPROVED** [review]
left some comments for cleanup in future PR ;)

**Review by akphi — APPROVED** [review]
left some comments for cleanup in future PR ;)

**Review by akphi — APPROVED** [review]

File: packages/legend-graph/src/graph/BasicModel.ts (hunk 1 above)
Comment (MauricioUyaguari) [reply packages/legend-graph/src/graph/BasicModel.ts:104]: this is not read only. it is set during the build graph phase

**Pull request merged**